packages/query-core/src/utils.ts

```diff
@@ -356,15 +356,13 @@ export function replaceData<
   } else if (options.structuralSharing !== false) {
     if (process.env.NODE_ENV !== 'production') {
       try {
-        JSON.stringify(prevData)
-        JSON.stringify(data)
+        replaceEqualDeep(prevData, data)
```

Comment:
I believe you can return here to avoid the double `replaceEqualDeep` calls.

Reply:
good catch!